context.go

```diff
@@ -184,6 +184,24 @@ func (c *Context) LookupTypeUnion(types []Type) *TypeUnion {
 	return typ
 }
 
+func (c *Context) LookupTypeOfValues(vals []Value) Type {
```

Comment:
I don't think I buy that this belongs here. I'd leave it in `expr/agg/collect.go` unless you plan to use it elsewhere, in which case putting it here seems fine for now.

Reply:
Yeah I'd prefer to keep this since I'm working on `into` and this will be needed there as well. Or perhaps it could go elsewhere? In `analyze.go` we duplicate this same logic but can't use it since values or not zed.Values. I'm also imaging this same problem will show up whenever you're constructing maps/arrays/sets.

Reply:
Let's keep it here for now then. I'm sure its proper home will eventually be apparent.